src/iterators.rs

```diff
@@ -601,11 +602,27 @@ macro_rules! custom_vec_iter_impl {
             fn __array__(
                 &self,
                 py: Python,
-                _dt: Option<&Bound<PyArrayDescr>>,
+                dtype: Option<PyObject>,
+                copy: Option<bool>,
             ) -> PyResult<PyObject> {
-                // Note: we accept the dtype argument on the signature but
-                // effictively do nothing with it to let Numpy handle the conversion itself
-                self.$data.convert_to_pyarray(py)
+                if copy == Some(false) {
+                    return Err(PyValueError::new_err(
+                        "A copy is needed to return an array from this object.",
+                    ));
+                }
+                let res = self.$data.convert_to_pyarray(py)?;
+                Ok(match dtype {
+                    Some(dtype) => {
+                        let locals = [("res", res), ("dtype", dtype)].into_py_dict_bound(py);
+                        py.eval_bound(
+                            "__import__('numpy').asarray(res, dtype=dtype)",
```

Comment:
I know this method exists but are there any alternatives? `__import__`  is strongly discouraged (https://docs.python.org/3/library/functions.html#import__)